Cargo.toml

```diff
@@ -391,13 +393,27 @@ zeroize = { version = "1.6.0", features = ["zeroize_derive", "std"] }
 zip = { version = "0.6.6", default-features = false, features = ["deflate","bzip2"] }
 zone = { version = "0.3", default-features = false, features = ["async"] }
 
+# NOTE: The test profile inherits from the dev profile, so settings under
+# profile.dev get inherited. AVOID setting anything under profile.test: that
+# will cause dev and test builds to diverge, which will cause more Cargo build
+# cache misses.
+
 [profile.dev]
-panic = "abort"
+# Note: This used to be panic = "abort" earlier, but that caused a lot of
+# duplicate dependency builds. Letting panic be "unwind" causes dependencies
+# across `cargo test` and `cargo run` to be unified. See omicron#4392.
+panic = "unwind"
+
 # See https://github.com/oxidecomputer/omicron/issues/4009 for some background context here.
 # By reducing the debug level (though keeping enough to have meaningful
 # backtraces), we reduce incremental build time and binary size significantly.
 debug = "line-tables-only"
 
+[profile.dev.build-override]
+# Setting this to line-tables-only results in a large improvement in build
+# times, because it allows target and host dependencies to be unified.
+debug = "line-tables-only"
```

Comment:
I understand why we want to set `line-tables-only` here, and I understand that we want it set for development builds. I also understand why we'd want to align development and test builds (e.g. with `panic = 'unwind'"` above).

Should we *also* be aligning `profile.dev` and `profile.dev.build-override` across other settings?

Should we, for example be setting `panic = "unwind"` for `build-override`?

Reply:
By default, [build-override for dev builds only sets](https://doc.rust-lang.org/cargo/reference/profiles.html#build-dependencies):

1. `opt-level = 0`. This is the same as the default opt-level for dev builds. We override the opt-level for some packages, but based on my testing it looks like those overrides are also carried through to build deps. If we ever set the global opt-level to something other than 0, however, we'll want to also set it for build-override.
* `codegen-units = 256`. In the unit graph the value appears to always be `null` so this doesn't appear to make a difference.
* `debug = false`. We're changing this setting as part of this PR.